### PR TITLE
[WNC-3] finish setting up migrations & refractor code

### PR DESCRIPTION
**Pull request by Tricker14**

### CODE REVIEW

**Review by VuKhoa23 — CHANGES_REQUESTED** [review]
update readme for makefile commands PLEASEE

File: main.go [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 	"github.com/VuKhoa23/advanced-web-be/startup"
 )
 
 func main() {
+	if len(os.Args) > 1 && os.Args[1] == "migrate" {
+		startup.Migrate()
+		return
+	}
+
 	startup.Execute()
```
Comment (VuKhoa23) [inline main.go]: I mentioned this before, please specify the command, not the number of arguments

Ex:
`if argument == migrate-up then do something else do something`

**Review by VuKhoa23 — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (VuKhoa23) [reply main.go]: DONT USE MAGIC NUMBER. Please be specific in your code

Comment (Tricker14) [reply main.go]: i already change from "migrate" to "migrate-up", imo using len(os.Args) > 1 && os.Args[1] == "migrate" is not magic at all, it just check the 1st argument of the command line to see what its type

Comment (Tricker14) [reply main.go]: using len(os.Args) > 1 && os.Args[1] == "migrate-up" is the only way to do i think so

Comment (VuKhoa23) [reply main.go]: `if len(argument) > 1`

Do you think anyone will understand what this line of code used for ? That why I said that code block is magic code

Comment (VuKhoa23) [reply main.go]: since until now we only have 1 command.  So maybe we can consider this as a tech debt and get back to it later

**Review by VuKhoa23 — APPROVED** [review]
LGTM !

File: main.go (hunk 1 above)
Comment (Tricker14) [reply main.go]: i already add a function in main to resolve it, go check it out

**Pull request merged**